ce/ce/amf/metadata-file.ts

```diff
@@ -17,37 +14,31 @@ import { Contacts } from './contact';
 import { DemandBlock, Demands } from './demands';
 import { GlobalSettings } from './global-settings';
 import { Info } from './info';
-import { Registries } from './registries';
-
-export class MetadataFile extends BaseMap implements Profile {
-  session!: Session;
+import { RegistriesDeclaration } from './registries';
 
-  private constructor(protected document: Document.Parsed, public readonly filename: string, public lineCounter: LineCounter, public readonly file: Uri, public readonly registry: Registry | undefined) {
+export class MetadataFile extends BaseMap {
+  private constructor(protected document: Document.Parsed, public readonly filename: string, public readonly file: Uri, public lineCounter: LineCounter, public readonly registryUri: Uri | undefined) {
     super(<YAMLMap<string, any>><any>document.contents);
-  }
 
-  async init(session: Session): Promise<MetadataFile> {
-    await this.demandBlock.init(session);
-    return this;
   }
 
-  static async parseMetadata(uri: Uri, session: Session, registry?: Registry): Promise<MetadataFile> {
-    return MetadataFile.parseConfiguration(uri.path, await uri.readUTF8(), session, registry);
+  static async parseMetadata(filename: string, uri: Uri, session: Session, registryUri?: Uri): Promise<MetadataFile> {
```

Comment:
Every call site for this function looks like:
```ts
MetadataFile.parseMetadata(x.fsPath, x, ...)
```
This argues to me that this convenience overload should just take the `uri` -- if a caller really wants to control the filename, they can `await uri.readUTF8()` and call the lower level one.

Secondly, `fsPath` is explicitly documented as not for display purposes:

>      * Returns a string representing the corresponding file system path of this URI.
     * Will handle UNC paths, normalizes windows drive letters to lower-case, and uses the
     * platform specific path separator.
     *
     * * Will *not* validate the path for invalid characters and semantics.
     * * Will *not* look at the scheme of this URI.
     * * The result shall *not* be used for display purposes but for accessing a file on disk.

Maybe we don't care and want to use it anyway, but it should be considered.

Reply:
This change is just getting that 'context' junk out of display operations and hoisting that problem out of this and into the immediate callers. I agree, the unwrapping into fsPath is a serious problem. I would like to see the over-reliance on URIs for things that aren't URIs burninated, but that would be its own very large but also very mechanical change so I don't want to combine it with this.

Reply:
Is https://github.com/microsoft/vcpkg-tool/pull/690/commits/dcff983adf751b29de88335015e36eb2f7722061 acceptable?

Reply:
Seems like a fine change to me